### PR TITLE
Redefine core function #'parse-cron-time-1

### PR DESCRIPTION
**Pull request by jcguu95**
See link for a detailed explanation.
https://github.com/endsec/scheduler/issues/3

---------------------------

`(parse-cron-entry "*/30 0 1-2,5-7 * */2 foo")` gives

`(:MINUTE (0 30) :HOUR (0) :DAY-OF-MONTH (1 2) :MONTH :EVERY :DAY-OF-WEEK
 (0 2 4 6))`

rather than

`(:MINUTE (0 30) :HOUR (0) :DAY-OF-MONTH (1 2 5 6 7) :MONTH :EVERY :DAY-OF-WEEK
 (0 2 4 6))`

### Reason

This is essentially because 
```
SCHEDULER-IMPLEMENTATION> (parse-cron-time-1 "1-2,5-7" '#.(alexandria:iota 10)) ;; => (1 2)
```

rather than `;; => (1 2 5 6 7)`.

### A solution

Redefine `parse-cron-time-1` by

```
(defun parse-cron-time-1 (spec range)
  (sort 
   (remove-duplicates
    (alexandria:flatten
     (loop for spec in (ppcre:split "," spec)
           collect (db (base . step) (ppcre:split "/" spec)
                     (let ((parse-result
                             (if (null step)
                                 (%parse-cron-time-1/no-step base range)
                                 (%parse-cron-time-1/step base range
                                                          (parse-integer (car step))))))
                       (assert (or (eql parse-result :every)
                                   (every (lambda (n) (member n range)) parse-result))
                               nil "PARSE-CRON-TIME-1: Each element of ~s must be a member of:~%~s."
                               parse-result (cons :every range))
                       parse-result)))))
   #'<))
```

Basically, I split the string by commas, and run what we had before over each term. Then collect. Then apply `sort`, `unique`, and `flatten`.

### Result

With the new definition,

`(parse-cron-time-1 "1-2,5-7" '#.(alexandria:iota 10))`,

and

```
SCHEDULER-IMPLEMENTATION> (parse-cron-entry "*/30 0 1-2,5-7 * */2 foo")
(:MINUTE (0 30) :HOUR (0) :DAY-OF-MONTH (1 2 5 6 7) :MONTH (:EVERY)
 :DAY-OF-WEEK (0 2 4 6))
"foo"
```

### CODE REVIEW

**Comment by jcguu95** [comment]
Is this PR desirable?

**Comment by dkochmanski** [comment]
I still need to look at it; sorry for poor responsiveness.

**Comment by dkochmanski** [comment]
thanks, I've resolved this problem by recursively parsing entries separated with `,` (that doesn't require changing higher structures). sorry for taking this long to resolve the issue.
